cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -218,26 +219,26 @@ def _add_resources(self):
             else:
                 instance_profile_name = self.custom_instance_role
         else:
-            resources_dependency_list.append(
+            resource_dependency_list.append(
                 self._add_default_instance_role(lambda_cleanup_policy_statements, build_tags_list)
             )
-            resources_dependency_list.append(
+            resource_dependency_list.append(
                 self._add_instance_profile(cleanup_policy_statements=lambda_cleanup_policy_statements)
             )
 
         self._add_imagebuilder_resources(
-            build_tags_map, ami_tags, instance_profile_name, lambda_cleanup_policy_statements, resources_dependency_list
+            build_tags_map, ami_tags, instance_profile_name, lambda_cleanup_policy_statements, resource_dependency_list
         )
 
         lambda_cleanup, permission, lambda_cleanup_execution_role = self._add_lambda_cleanup(
             lambda_cleanup_policy_statements, build_tags_list
         )
-        resources_dependency_list.extend([lambda_cleanup, permission])
+        resource_dependency_list.extend([lambda_cleanup, permission])
```

Comment:
If the lambda_log is a resource, we should add it to resource_dependency_list to make sure it is deleted before the role

Reply:
added